File: 11.盛最多水的容器/11.php
<?php
class Solution {

    /**
     * @param Integer[] $height
     * @return Integer
     */
    function maxArea($height) {
        $length = count($height);
        $s = 0;
        for ($i = 0, $j = $length - 1; $i < $j; true) {
            $temp_s = ($j - $i) * min($height[$i], $height[$j]);

            if ($temp_s > $s) {
                $s = $temp_s;
            }

            $height[$i] > $height[$j] ? $j-- : $i++;
        }

        return $s;
    }
}

File: 11.盛最多水的容器/11.go
package main

func maxArea(height []int) int {
	length := len(height)

	var s int
	var i int

	for j:=length-1; i < j ; {
		temp_s := (j - i) * min(height[i], height[j])

		if (temp_s > s) {
			s = temp_s
		}

		if height[i] > height[j] {
			j--
		} else {
			i++
		}
	}

	return s
}

func min(a, b int) int {
	if b < a {
		return b
	}
	return a
}